src/gwt/src/org/rstudio/studio/client/workbench/prefs/views/AppearancePreferencesPane.java

```diff
@@ -185,23 +179,72 @@ public void onChange(ChangeEvent event)
       theme_.getListBox().getElement().getStyle().setHeight(225, Unit.PX);
       theme_.getListBox().addChangeHandler(new ChangeHandler()
       {
+         @Override
          public void onChange(ChangeEvent event)
          {
-            preview_.setTheme(themeList_.get(theme_.getValue()).getUrl());
+            AceTheme aceTheme = themeList_.get(theme_.getValue());
+            preview_.setTheme(aceTheme.getUrl());
+            removeThemeButton_.setEnabled(!aceTheme.isDefaultTheme());
          }
       });
       theme_.addStyleName(res.styles().themeChooser());
    
+      AceTheme currentTheme = uiPrefs_.theme().getGlobalValue();
+      addThemeButton_ = new ThemedButton("Add...", event ->
+         fileDialogs.openFile(
+            "Theme Files (*.tmTheme *.rstheme)",
+            RStudioGinjector.INSTANCE.getRemoteFileSystemContext(),
+            workbenchContext.getCurrentWorkingDir(),
+            "Theme Files (*.tmTheme *.rstheme)",
+            (input, indicator) ->
+            {
+               if (input == null)
+                  return;
+               
+               String inputStem = input.getStem();
+               String inputPath = input.getPath();
+               boolean isTmTheme = StringUtil.equalsIgnoreCase(".tmTheme", input.getExtension());
+               boolean found = false;
+               for (AceTheme theme: themeList_.values())
+               {
+                  if (theme.isLocalCustomTheme() &&
+                     StringUtil.equalsIgnoreCase(theme.getFileStem(), inputStem))
+                  {
+                     showThemeExistsDialog(inputStem, () -> addTheme(inputPath, themes, isTmTheme));
+                     found = true;
+                     break;
+                  }
+               }
+               
+               if (!found)
+               {
+                  addTheme(inputPath, themes, isTmTheme);
+               }
+               
+               indicator.onCompleted();
+            }));
+      addThemeButton_.setLeftAligned(true);
+      removeThemeButton_ = new ThemedButton(
+         "Remove...",
```

Comment:
Nit: By convention (in buttons and menus) `...` indicates an action that requires user input (i.e. launches a dialog) vs. one that takes place immediately; "Add" should have the `...`, but "Remove" shouldn't.